Framework/PythonInterface/plugins/algorithms/SaveYDA.py

```diff
@@ -0,0 +1,316 @@
+
+            if not isinstance(ws, Workspace2D):
+                issues["InputWorkspace"] = "Input Workspace is not a Workspace2D"
+
+        return issues
+
+    def PyExec(self):
+        """ Main execution body
+        """
+        # Properties
+        ws = mtd[self.getPropertyValue("InputWorkspace")]
+        filename = self.getProperty("Filename").value
+
+        # check workspace exists
+        if not ws:
+            raise NotImplementedError("InputWorkspace does not exist")
+
+        run = ws.getRun()
+        ax = ws.getAxis(1)
+        nHist = ws.getNumberHistograms()
+
+        # check sample logs exists
+        if len(run.getLogData()) == 0:
+            raise NotImplementedError("No sample log data exist in workspace: "
+                                      + self.getPropertyValue("InputWorkspace"))
+
+        # save sample log data in lists, commented sequences an commented maps
+        # commented sequences and maps are used to keep Data in the order they get inserted
+        # if a log does not exist a warning is written on the log and the data is not saved in the file
+
+        metadata = OrderedDict()
+
+        metadata["format"] = "yaml/frida 2.0"
+        metadata["type"] = "generic tabular data"
+
+        hist = []
+
+        if run.hasProperty("proposal_number"):
+            propn = "Proposal number " + run.getLogData("proposal_number").value
+            hist.append(propn)
+        else:
+            self.log().warning("no proposal number found")
+
+        if run.hasProperty("proposal_title"):
+            propt = run.getLogData("proposal_title").value
+            hist.append(propt)
+        else:
+            self.log().warning("no proposal title found")
+
+        if run.hasProperty("experiment_team"):
+            expt = run.getLogData("experiment_team").value
+            hist.append(expt)
+        else:
+            self.log().warning("no experiment team found")
+
+        hist.append("data reduced with mantid")
+
+        rpar = []
+
+        if run.hasProperty("temperature"):
+            temperature = float(run.getLogData("temperature").value)
+
+            temp = OrderedDict()
+            temp["name"] = "T"
+            temp["unit"] = "K"
+            temp["val"] = temperature
+            temp["stdv"] = 0
+
+            rpar.append(temp)
+        else:
+            self.log().warning("no temperature found")
+
+        if run.hasProperty("Ei"):
+            eimeV = float(run.getLogData("Ei").value)
+
+            ei = OrderedDict()
+            ei["name"] = "Ei"
+            ei["unit"] = "meV"
+            ei["val"] = eimeV
+            ei["stdv"] = 0
+
+            rpar.append(ei)
+        else:
+            self.log().warning("no Ei found")
+
+        coord = OrderedDict()
+
+        x = FlowOrderedDict()
+
+        x["name"] = "w"
+        x["unit"] = "meV"
+
+        # set_flow style is used to keep the Frida 2.0 yaml format
+        coord["x"] = x
+        #coord["x"].fa.set_flow_style()
```

Comment:
If this line is not valid please can it be removed to avoid confusion in the future?